File: backend/app/schemas/audit.py
from typing import Optional, List
from datetime import datetime, date
from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: int
    user_id: Optional[int] = None
    username: Optional[str] = None
    action: str
    resource_type: Optional[str] = None
    resource_id: Optional[str] = None
    detail: Optional[str] = None
    ip_address: Optional[str] = None
    user_agent: Optional[str] = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class AuditLogQuery(BaseModel):
    username: Optional[str] = None
    action: Optional[str] = None
    resource_type: Optional[str] = None
    status: Optional[str] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    page: int = 1
    page_size: int = 20


class AuditLogListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    items: List[AuditLogResponse]

File: frontend/src/api/audit.ts
import request from './request'

export interface AuditLogRecord {
  id: number
  user_id: number | null
  username: string | null
  action: string
  resource_type: string | null
  resource_id: string | null
  detail: string | null
  ip_address: string | null
  user_agent: string | null
  status: string
  created_at: string
}

export interface AuditLogParams {
  page?: number
  page_size?: number
  username?: string
  action?: string
  resource_type?: string
  status?: string
  start_date?: string
  end_date?: string
}

export interface AuditLogListResult {
  total: number
  page: number
  page_size: number
  items: AuditLogRecord[]
}

export function getAuditLogListApi(params: AuditLogParams) {
  return request.get<unknown, AuditLogListResult>('/audit-logs', { params })
}
